kegen/gboy: Fix typo in readme.md; add labels to info.json and reformat for readability (#18269)
qmk/qmk_firmware: keyboards/kegen/gboy/readme.md

* Keyboard Maintainer: [Matt Chan](https://github.com/mattchan)
* Hardware Supported: Kegen G-Boy

> * Hardware Availability: Unfortunately, there is no site for this, and the Kegen Discord is a ghost town. The group buy runners disappeared after round two units were delivered.

Author: Does this second sentence violate the [Code of Conduct](https://github.com/qmk/.github/blob/3eb6438286a633b2c79760100369e4035bc95399/CODE_OF_CONDUCT.md)? It's a slight towards the group buy runners, but my intention here is provide some more context regarding the circumstances of this board's firmware.

Reviewer: I would recommend this sort of wording:
```suggestion
* Hardware Availability: Unfortunately, there is no site for this, and the Kegen Discord is no longer active. The group buy runners appear to have disappeared after round two units were delivered.
```

Author: Sounds good. I've updated the wording.
